src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -206,9 +277,49 @@ trait ScriptGenerators extends BitcoinSLogger {
     (p2SHScriptSignature, p2SHScriptPubKey, privateKeys)
   }
 
+  /**
+    * Generates a signed [[CLTVScriptSignature]] that spends from a [[CLTVScriptPubKey]] correctly
+    *
+    * @return the signed [[CLTVScriptSignature]], the [[CLTVScriptPubKey]] it spends, and the sequences of [[ECPrivateKey]]
+    *         used to sign the scriptSig
+    */
+  def signedCLTVScriptSignature(scriptNumber : ScriptNumber, lockTime : UInt32, sequence : UInt32) : Gen[(ScriptSignature, CLTVScriptPubKey, Seq[ECPrivateKey])] = for {
```

Comment:
I think this should technically return something of type `CLTVScriptSignature` instead of just a generic `ScriptSignature`